10up/classifai
# feat: add crop command

### Description of the Change
This PR add new `crop` command to "smart crop" uploaded images. This command output detailed error about the whole process which help trouble shooting easier and more straight forward.

Some example of errors outputed:
```
Fetching images ...
Fetching images ... DONE (1)
Cropping 1 images ...  100% [==================================================] 0:02 / 0:02
1842:150x150: The value '1' is not valid. (classifai_smart_cropping_api_validation_errors).
Error: Cropped 0 images, 1 errors.
```

```
Fetching images ...
Fetching images ... DONE (1)
Cropping 1 images ...  100% [==================================================] 0:01 / 0:02
1842:150x150: Image URL is not accessible. (InvalidImageUrl).
Error: Cropped 0 images, 1 errors.
```

```
$ wp classifai crop
Error: Smart Cropping is disabled. Enable it in Image settings to use this command.
```

### Verification Process
Please test this PR on a live server without CND plugin enabled.
1. Disable Smart Cropping.
2. Upload an image.
3. See the thumbnail is not "smart cropped".
4. Run `wp classifai crop`.
5. See error "Smart Cropping is disabled.".
6. Enable Smart Cropping.
7. Run `wp classifai crop` again
8. Due to the API error which is addressing in #229, you will get api validation error. Apply the fix on that PR, you will see the thumbnail get cropped correctly.


### Checklist:






### Applicable Issues
Fixes #176


### Changelog Entry
